core/src/main/scala/org/scalacheck/ops/SeededGen.scala

```diff
@@ -0,0 +1,128 @@
+  *       your generator depends on the value of the seed provided (passed along implicitly).
+  *       Using a plain [[Gen]] will work as expected in all other contexts.
+  *
+  * {{{
+  *   trait ID extends SeededGen.Tag
+  *   val genNamespacedID = SeededGen[String, ID, String] { implicit id =>
+  *     for {
+  *       inner <- Gen.identifier // doesn't need to be a SeededGen
+  *     } yield s"thing-$id-$inner"
+  *   }
+  *   assert(gen.instance("1") startsWith "thing-1-")
+  * }}}
+  *
+  * @note You can treat a [[SeededGen]] as a [[Gen]] when inside the definition of a [[SeededGen]] if you
+  *       make the given seed implicit.
+  *
+  * {{{
+  *   val gen = SeededGen[String, ID, Map[String, String]] { implicit id =>
+  *     for {
+  *       namespacedID <- genNamespacedID // implicitly converted from SeededGen to Gen because of the implicit id
+  *       rank <- Gen.choose(1, 10)
+  *     } yield Map(
+  *       "id" -> namespacedID,
+  *       "rank" -> rank.toString
+  *     )
+  *   }
+  * }}}
+  *
+  * Typically, it is most convenient to define a subclass of [[SeededGen]] so that you can benefit from
+  * type inference and hide the `SeededGen.Tag` type.
+  *
+  * {{{
+  *   class IDSeededGen[V](genFn: String @@ IDSeededGen.Tag => Gen[V])
+  *     extends SeededGen[IDSeededGen.Tag, String, V](genFn)
+  *
+  *   object IDSeededGen {
+  *     trait Tag
+  *     type GenFn[V] = SeededGen.GenFn[Tag, String, V]
+  *     def apply[V](genFn: GenFn[V]): IDSeededGen[V] = new IDSeededGen(genFn)
+  *   }
+  *
+  *   val gen = IDSeededGen { implicit id =>
+  *     for {
+  *       namespacedID <- genNamespacedID
+  *       rank <- Gen.choose(1, 10)
+  *     } yield Map(
+  *       "id" -> namespacedID,
+  *       "rank" -> rank.toString
+  *     )
+  *   }
+  * }}}
+  *
+  * @param genFn a function from seed (typically implicit) to a generator
+  * @tparam S the seed type, must have an implicit [[SeedExtractor]] to mix in with the starting seed
+  * @tparam T the type of [[org.scalacheck.ops.SeededGen.Tag]] used to avoid exposing
+  *           implicit type arguments on primitive values
+  * @tparam V the type of value produced by the generator
+  */
+class SeededGen[-S, T <: SeededGen.Tag, V : ru.TypeTag](
+  genFn: S @@ T => Gen[V]
+)(implicit
+  extractor: SeedExtractor[S],
+  extractorTag: SeedExtractor[ru.TypeTag[V]]
+) {
+
+  /** Stateless function that generates an identical [[Gen]] of V instances for a given seed and [[GenConfig]]. */
+  def gen(seed: S): Gen[V] = genFn(seed.taggedWith[T])
+
+  /** Stateless function that generates an identical instance value V for a given seed and [[GenConfig]]. */
+  def instance(seed: S)(implicit c: GenConfig): V = withSeed(seed)
+
+  /** Creates a new seeded gen */
+  def compose[U : ru.TypeTag](fn: Gen[V] => Gen[U]): SeededGen[S, T, U] = {
+    new SeededGen[S, T, U](seed => fn(gen(seed)))
+  }
+
+  protected def withSeed(seed: S)(implicit c: GenConfig): V = {
```

Comment:
Any reason not to just move this to `instance`? Also, why not make the class `final` to encourage using the Companion / type alias style?